cocos/base/CCDirector.cpp

```diff
@@ -474,7 +515,27 @@ void Director::popMatrix(MATRIX_STACK_TYPE type)
     }
     else if(MATRIX_STACK_TYPE::MATRIX_STACK_PROJECTION == type)
     {
-        _projectionMatrixStack.pop();
+        _projectionMatrixStackList[0].pop();
+    }
+    else if(MATRIX_STACK_TYPE::MATRIX_STACK_TEXTURE == type)
+    {
+        _textureMatrixStack.pop();
+    }
+    else
+    {
+        CCASSERT(false, "unknown matrix stack type");
+    }
+}
+
+void Director::popMatrix(MATRIX_STACK_TYPE type, unsigned int index)
```

Comment:
`_modelViewMatrixStack` and  `_textureMatrixStack ` is not need to support this function, so i think it is better to rename it to `popProjectionMatrix(unsigned index)`. The same opinion with `loadIdentityMatrix ` and other functions.